modules/java build issues

When trying to compile syslog-ng with `make -j4`, I got the following results:

```
 GEN      modules/java/org/syslog_ng/LogTemplate.class
../modules/java/SyslogNgClassLoader.java:69: error: cannot find symbol
      InternalMessageSender.error("Exception: " + e.getMessage());
      ^
  symbol:   variable InternalMessageSender
  location: class SyslogNgClassLoader
../modules/java/SyslogNgClassLoader.java:73: error: cannot find symbol
      InternalMessageSender.error("Error: " + e.getMessage());
      ^
  symbol:   variable InternalMessageSender
  location: class SyslogNgClassLoader
../modules/java/SyslogNgClassLoader.java:77: error: cannot find symbol
      InternalMessageSender.error("Error while expanding path list: " + e.getMessage());
      ^
  symbol:   variable InternalMessageSender
  location: class SyslogNgClassLoader
../modules/java/SyslogNgClassLoader.java:129: error: cannot find symbol
        InternalMessageSender.debug("Add path to classpath: " + path);
        ^
  symbol:   variable InternalMessageSender
  location: class SyslogNgClassLoader
  CCLD     examples/amqp_producer
4 errors
Makefile:12044: recipe for target 'modules/java/org/syslog_ng/SyslogNgClassLoader.class' failed
make: *** [modules/java/org/syslog_ng/SyslogNgClassLoader.class] Error 1
make: *** Waiting for unfinished jobs....
```

Trying to track things down, I noticed a few more things:

* Unlike most (if not all) other modules, `modules/java/Makefile.am` adds `libmod-java.la` to `module_LTLIBRARIES`, instead of `modules/java/libmod-java.la`. It also doesn't have the `mod-java` and `modules/java` shortcut targets, which would have made it a minute faster to arrive to the next problem:

```
$ make libmod-java.la
  YACC     modules/java/native/java-grammar.c
  CC       modules/java/native/libmod_java_la-java-grammar.lo
  CC       modules/java/native/libmod_java_la-java-plugin.lo
  CC       modules/java/native/libmod_java_la-java-destination.lo
  CC       modules/java/proxies/libmod_java_la-java-destination-proxy.lo
In file included from ../modules/java/proxies/java-destination-proxy.c:26:0:
../modules/java/proxies/java-logmsg-proxy.h:28:24: fatal error: LogMessage.h: No such file or directory
 #include "LogMessage.h"
                        ^
compilation terminated.
Makefile:8082: recipe for target 'modules/java/proxies/libmod_java_la-java-destination-proxy.lo' failed
```

In short, I had trouble compiling the Java plugin alone, without having to go through the rest. So much so, that I gave up and just disabled java for the time being.

--- a/modules/java/Makefile.am
+++ b/modules/java/Makefile.am
@@ -1,5 +1,7 @@
 if ENABLE_JAVA
 
+modules/java mod-java: modules/java/libmod-java.la
+
 module_LTLIBRARIES += modules/java/libmod-java.la
 
 modules_java_libmod_java_la_CFLAGS = \
@@ -124,6 +126,10 @@ java-clean-hook:
 INSTALL_EXEC_HOOKS += java-install-exec-hook
 UNINSTALL_HOOKS += java-uninstall-exec-hook
 CLEAN_HOOKS += java-clean-hook
+
+else
+# when Java is disabled
+modules/java mod-java:
 
 endif
 

--- a/modules/java/Makefile.am
+++ b/modules/java/Makefile.am
@@ -1,10 +1,8 @@
-export top_srcdir
-
 if ENABLE_JAVA
 
-module_LTLIBRARIES += libmod-java.la
+module_LTLIBRARIES += modules/java/libmod-java.la
 
-libmod_java_la_CFLAGS = \
+modules_java_libmod_java_la_CFLAGS = \
     $(JNI_CFLAGS)  \
     $(INCUBATOR_CFLAGS) \
     -I$(top_srcdir)/modules/java    \
@@ -12,7 +10,7 @@ libmod_java_la_CFLAGS = \
     -I$(top_srcdir)/modules/java/native \
     -I$(top_builddir)/modules/java/native
 
-libmod_java_la_SOURCES = \
+modules_java_libmod_java_la_SOURCES = \
     modules/java/native/java-grammar.y \
     modules/java/native/java-plugin.c \
     modules/java/native/java-destination.c \
@@ -31,13 +29,13 @@ libmod_java_la_SOURCES = \
     modules/java/native/java_machine.c \
     modules/java/native/java_machine.h
 
-libmod_java_la_LIBADD = \
+modules_java_libmod_java_la_LIBADD = \
     $(JNI_LIBS) $(INCUBATOR_LIBS)
 
-libmod_java_la_LDFLAGS = \
+modules_java_libmod_java_la_LDFLAGS = \
     -avoid-version -module -no-undefined
 
-libmod_java_la_DEPENDENCIES  = \
+modules_java_libmod_java_la_DEPENDENCIES  = \
         $(MODULE_DEPS_LIBS)
 
 JAVA_PKG_NAME = org.syslog_ng
@@ -83,8 +81,35 @@ modules/java/SyslogNg.jar: $(JAVA_CLASS_FILES) $(JAVA_HEADER_FILES)
 $(JAVA_HEADER_FILES): $(JAVA_CLASS_FILES)
 	$(AM_V_GEN) $(JAVAH) -cp ./ -o $@ $(JAVA_PKG_NAME).$(notdir $(basename $@))
 
-$(JAVA_CLASS_PATH)/%.class: $(top_srcdir)/modules/java/%.java $(JAVA_FILES)
-	$(AM_V_GEN) $(JAVAC) -d ./ -cp ./ $<
+$(JAVA_CLASS_PATH)/InternalMessageSender.class: $(JAVA_SRC_DIR)/InternalMessageSender.java
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/SyslogNgClassLoader.class: $(JAVA_SRC_DIR)/SyslogNgClassLoader.java $(JAVA_CLASS_PATH)/InternalMessageSender.class
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/LogMessage.class: $(JAVA_SRC_DIR)/LogMessage.java
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/LogTemplate.class: $(JAVA_SRC_DIR)/LogTemplate.java
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/LogPipe.class: $(JAVA_SRC_DIR)/LogPipe.java $(JAVA_CLASS_PATH)/InternalMessageSender.class
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/LogDestination.class: $(JAVA_SRC_DIR)/LogDestination.java $(JAVA_CLASS_PATH)/LogPipe.class
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/TextLogDestination.class: $(JAVA_SRC_DIR)/TextLogDestination.java $(JAVA_CLASS_PATH)/LogDestination.class
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/StructuredLogDestination.class: $(JAVA_SRC_DIR)/StructuredLogDestination.java $(JAVA_CLASS_PATH)/LogDestination.class
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/DummyStructuredDestination.class: $(JAVA_SRC_DIR)/DummyStructuredDestination.java $(JAVA_CLASS_PATH)/StructuredLogDestination.class
+	$(JAVAC) -d ./ -cp ./ $<
+
+$(JAVA_CLASS_PATH)/DummyTextDestination.class: $(JAVA_SRC_DIR)/DummyTextDestination.java $(JAVA_CLASS_PATH)/TextLogDestination.class
+	$(JAVAC) -d ./ -cp ./ $<
 
 java-install-exec-hook:
 	$(mkinstalldirs) $(DESTDIR)$(moduledir)

